### ISSUE TITLE
Gradient of LazyOp containing a test function

### ISSUE THREAD

**Issue by bmxam** [bcube-project/Bcube.jl#34]
The following example fails, I don't know if it's hard to fix...

```julia
module mwe
using Bcube
using StaticArrays
using LinearAlgebra

mesh = one_cell_mesh(:quad)
dΩ = Measure(CellDomain(mesh), 2)

U = TrialFESpace(FunctionSpace(:Lagrange, 1), mesh; size = 2)
V = TestFESpace(U)

p = PhysicalFunction(x -> 1.0)
c = CellInfo(mesh, 1)

assemble_linear(v -> ∫(∇(p * v) ⋅ SA[1, 0])dΩ, V)

end
```

**Comment by ghislainb** [comment]
The following expression works with v0.1.3 (and with `size=1` for `U`):
```julia
v -> ∫(2 * ∇(v) ⋅ SA[1, 0])dΩ
```
Currently, it is not recommended (and not well supported!) to combine operations on a test function `v` before taking its gradient because there is a specific processing with MapOver on `∇(v)`. For `CellFunction`, the gradient of combined operations is supported and there should be no problem.
For example:
```julia
u = FEFunction(U)
a = assemble_linear(v -> ∫(∇(v) ⋅ ∇(2 * u))dΩ, V)
b = assemble_linear(v -> ∫(∇(v) ⋅ ∇(u))dΩ, V)
@assert all(a .≈ 2 .* b) # true
```
Is this ok for you?

**Comment by bmxam** [comment]
> The following expression works with v0.1.3 (and with `size=1` for `U`):
> 
> ```julia
> v -> ∫(2 * ∇(v) ⋅ SA[1, 0])dΩ
> ```
> 
> Currently, it is not recommended (and not well supported!) to combine operations on a test function `v` before taking its gradient because there is a specific processing with MapOver on `∇(v)`. For `CellFunction`, the gradient of combined operations is supported and there should be no problem. For example:
> 
> ```julia
> u = FEFunction(U)
> a = assemble_linear(v -> ∫(∇(v) ⋅ ∇(2 * u))dΩ, V)
> b = assemble_linear(v -> ∫(∇(v) ⋅ ∇(u))dΩ, V)
> @assert all(a .≈ 2 .* b) # true
> ```
> 
> Is this ok for you?

Yes it's ok for now. Maybe we could keep this issue opened with the "enhancement" label?
